src/test/rpc/Simulate_test.cpp

```diff
@@ -0,0 +1,819 @@
+                auto result = resp[jss::result];
+                checkBasicReturnValidity(
+                    result, tx, 1, env.current()->fees().base);
+
+                BEAST_EXPECT(result[jss::engine_result] == "tesSUCCESS");
+                BEAST_EXPECT(result[jss::engine_result_code] == 0);
+                BEAST_EXPECT(
+                    result[jss::engine_result_message] ==
+                    "The simulated transaction would have been applied.");
+
+                if (BEAST_EXPECT(
+                        result.isMember(jss::meta) ||
+                        result.isMember(jss::meta_blob)))
+                {
+                    Json::Value metadata;
+                    if (result.isMember(jss::meta_blob))
+                    {
+                        auto unHexed =
+                            strUnHex(result[jss::meta_blob].asString());
+                        SerialIter sitTrans(makeSlice(*unHexed));
+                        metadata = STObject(std::ref(sitTrans), sfGeneric)
+                                       .getJson(JsonOptions::none);
+                    }
+                    else
+                    {
+                        metadata = result[jss::meta];
+                    }
+                    if (BEAST_EXPECT(
+                            metadata.isMember(sfAffectedNodes.jsonName)))
+                    {
+                        BEAST_EXPECT(
+                            metadata[sfAffectedNodes.jsonName].size() == 1);
+                        auto node = metadata[sfAffectedNodes.jsonName][0u];
+                        if (BEAST_EXPECT(
+                                node.isMember(sfModifiedNode.jsonName)))
+                        {
+                            auto modifiedNode = node[sfModifiedNode.jsonName];
+                            BEAST_EXPECT(
+                                modifiedNode[sfLedgerEntryType.jsonName] ==
+                                "AccountRoot");
+                            auto finalFields =
+                                modifiedNode[sfFinalFields.jsonName];
+                            BEAST_EXPECT(
+                                finalFields[sfDomain.jsonName] == newDomain);
+                        }
+                    }
+                    BEAST_EXPECT(metadata[sfTransactionIndex.jsonName] == 0);
+                    BEAST_EXPECT(
+                        metadata[sfTransactionResult.jsonName] == "tesSUCCESS");
+                }
+            };
+
+            Json::Value tx;
+
+            tx[jss::Account] = env.master.human();
+            tx[jss::TransactionType] = jss::AccountSet;
+            tx[sfDomain.jsonName] = newDomain;
+
+            // test with autofill
+            testTx(env, tx, validateOutput);
+
+            tx[sfSigningPubKey.jsonName] = "";
+            tx[sfTxnSignature.jsonName] = "";
+            tx[sfSequence.jsonName] = 1;
+            tx[sfFee.jsonName] = "12";
```

Comment:
Please use 
`env.current()->fees()` and its members here and below 
for example 
```
drops(env.current()->fees().increment)  - for account delete
env.current()->fees().accountReserve(0) - 1   for maximum amount which still can't create account
```
and so on